src/main/java/com/codeless/api/automation/constant/PricingPlan.java

```diff
@@ -10,24 +10,33 @@
 @RequiredArgsConstructor
 public enum PricingPlan {
 
-  FREE(20),
-  BASIC(40),
-  PROFESSIONAL(100);
+  DEFAULT("default", 20,40, 100),
+  EXECUTIONS("/api/executions", 10,20, 50);
 
-  private final int bucketCapacity;
+  private final String requestPath;
+  private final int freeCapacity;
+  private final int basicCapacity;
+  private final int proCapacity;
 
-  public Bandwidth getLimit() {
-    return Bandwidth.builder().capacity(bucketCapacity)
-        .refillIntervally(bucketCapacity, Duration.ofHours(1)).build();
-  }
-
-  public static PricingPlan resolvePricingFromUserPlan(UserPlan userPlan) {
+  public Bandwidth getLimit(UserPlan userPlan){
     if (UserPlan.BASIC.equals(userPlan)) {
-      return BASIC;
+      return Bandwidth.builder().capacity(basicCapacity)
+          .refillIntervally(basicCapacity, Duration.ofMinutes(50)).build();
```

Comment:
per hour, is not going to fly, it is too much.
I am ok to have per hour for executions, in particular. createExecution should not exceed 12 executions per hours. Execution is really what we protect because it is compute for which we pay.

For other APIs, it is ok if it is per second. The same numbers you have, but per seconds.